app/src/main/java/com/github/fribourgsdp/radio/Playlist.kt

```diff
@@ -0,0 +1,44 @@
+class Playlist (playListName: String, songSet: Set<Song>, genreType: Genre){
+    var name: String = playListName
+    private val songs: MutableSet<Song>
+    var genre: Genre = genreType
+
+    init {
+        val result = mutableSetOf<Song>()
+        for (song in songSet){
+            result.add(Song(song.name, song.artist, song.lyrics))
+        }
+        songs = result
+    }
+    constructor(playListName: String, genreType: Genre): this(playListName, mutableSetOf(), genreType)
+    constructor(playListName: String): this(playListName, mutableSetOf(), Genre.NONE)
+
+
+    fun addSong(song: Song){
+        songs.add(song)
+    }
+
+    fun addSongs(addedSongs: Set<Song>){
+        songs.addAll(addedSongs)
+    }
+
+    fun removeSong(song: Song){
+        songs.remove(song)
+    }
+
+    fun removeSongs(removedSongs: Set<Song>){
+        songs.removeAll(removedSongs)
+    }
+
+    fun combinePlaylists(other: Playlist, newName: String = name, newGenre: Genre = Genre.NONE) {
+        name = newName
+        genre = newGenre
+        for (song in other.songs) {
```

Comment:
Same here, why not use 'addAll' ?

Reply:
Same as above, I did this to avoid the fact that if the owner of the 'other' playlist changes a song name, it doesn't change in our new playlist that we combined previously.